dnf/dnf-utils.c

```diff
@@ -19,6 +19,40 @@
  */
 
 #include "dnf-utils.h"
+#include "libsmartcols/libsmartcols.h"
+
+
+// transaction details columns
+enum { COL_NEVRA, COL_REPO, COL_SIZE };
+
+
+static gint
+dnf_package_cmp_cb (DnfPackage **pkg1, DnfPackage **pkg2)
+{
+  return dnf_package_cmp (*pkg1, *pkg2);
+}
+
+
+gboolean
+dnf_utils_add_transaction_packages (struct libscols_table *tb, GPtrArray *pkgs)
+{
+  // sort packages by NEVRA
+  g_ptr_array_sort (pkgs, (GCompareFunc) dnf_package_cmp_cb);
+
+  struct libscols_line *ln;
+  for (guint i = 0; i < pkgs->len; i++)
+    {
+      DnfPackage *pkg = pkgs->pdata[i];
+
+        ln = scols_table_new_line(tb, NULL);
+        gchar * nevra = g_strconcat(" ", dnf_package_get_nevra (pkg));
+        scols_line_set_data(ln, COL_NEVRA, nevra);
+        g_free(nevra);
```

Comment:
you could use SCOLS_FL_TREE to column and make all packages as a child to action